src/main/java/com/commercetools/sync/products/utils/ProductVariantUpdateActionUtils.java

```diff
@@ -126,10 +135,51 @@
         @Nonnull final ProductVariant oldProductVariant,
         @Nonnull final ProductVariantDraft newProductVariant) {
         //TODO: Right now it always builds SetPrices UpdateAction, comparison should be calculated GITHUB ISSUE#101.
-        final List<PriceDraft> newProductVariantPrices =
-            newProductVariant.getPrices() == null ? new ArrayList<>() : newProductVariant.getPrices();
-        final SetPrices setPricesUpdateAction = SetPrices.of(oldProductVariant.getId(), newProductVariantPrices);
-        return Collections.singletonList(setPricesUpdateAction);
+        final SetPrices setPricesUpdateAction = SetPrices.of(oldProductVariant.getId(),
+            emptyIfNull(newProductVariant.getPrices()));
+        return singletonList(setPricesUpdateAction);
+    }
+
+    /**
+     * Compares the {@link List} of {@link AssetDraft}s of a {@link ProductVariantDraft} and a
+     * {@link ProductVariant} and returns a {@link List} of {@link UpdateAction}&lt;{@link Product}&gt;. If both the
+     * {@link ProductVariantDraft} and the {@link ProductVariant} have identical list of assets, then no update action
+     * is needed and hence an empty {@link List} is returned. In case, the new product variant draft has a list of
+     * assets in which a duplicate key exists, the error callback is triggered and an empty list is returned.
+     *
+     * @param oldProductVariant the {@link ProductVariant} which should be updated.
+     * @param newProductVariant the {@link ProductVariantDraft} where we get the new list of assets.
+     * @param syncOptions       responsible for supplying the sync options to the sync utility method. It is used for
+     *                          triggering the error callback within the utility, in case of errors.
+     * @return a list that contains all the update actions needed, otherwise an empty list if no update actions are
+     *         needed.
+     */
+    @Nonnull
+    public static List<UpdateAction<Product>> buildProductVariantAssetsUpdateActions(
+        @Nonnull final ProductVariant oldProductVariant,
+        @Nonnull final ProductVariantDraft newProductVariant,
+        @Nonnull final ProductSyncOptions syncOptions) {
+
+        try {
+            return buildAssetsUpdateActions(
+                oldProductVariant.getAssets(),
+                newProductVariant.getAssets(),
+                (oldAsset, newAssetDraft) ->
+                    buildActions(oldProductVariant.getId(), oldAsset, newAssetDraft, syncOptions),
+                key ->
+                    RemoveAsset.ofVariantIdWithKey(oldProductVariant.getId(), key, true),
+                (newAssetOrder) ->
+                    ChangeAssetOrder.ofVariantId(oldProductVariant.getId(), newAssetOrder, true),
+                (assetDraft, index) ->
+                    AddAsset.ofVariantId(oldProductVariant.getId(), assetDraft)
+                            .withPosition(index)
+                            .withStaged(true));
```

Comment:
1. Regarding changing the width of the screen checkstyle issue I agree also, maybe we could increase it a little bit. But I would rather do it separately since I would need a little more time for it to investigate the best width and adjust the classes which are split over several lines. https://github.com/commercetools/commercetools-sync-java/issues/264
2. Already addressed: 
9d5dc63